packages/types/src/utils/executionAddress.ts

```diff
@@ -0,0 +1,48 @@
+import {keccak256} from "ethereum-cryptography/keccak.js";
+import {ByteVectorType} from "@chainsafe/ssz";
+
+export type ByteVector = Uint8Array;
+
+export class ExecutionAddressType extends ByteVectorType {
+  constructor() {
+    super(20, {typeName: "ExecutionAddress"});
+  }
+  toJson(value: ByteVector): unknown {
+    const str = super.toJson(value) as string;
+    return toChecksumAddress(str);
+  }
+}
+
+function isAddressValid(address: string): boolean {
+  return /^(0x)?[0-9a-f]{40}$/i.test(address);
+}
+
+/**
+ * Formats an address according to [ERC55](https://eips.ethereum.org/EIPS/eip-55)
+ */
+export function toChecksumAddress(address: string): string {
```

Comment:
is this used anywhere yet? or is this PR just adding the functionality for later?

Reply:
Do you mean specifically because of the `export` keyword?

Reply:
I guess I missed L64 of `primitive/sszTypes.ts` where this is being used.

This is fine to add here for now.

Reply:
It also needs to be exported for unit testing

Reply:
> This is fine to add here for now.

yeah, thought so as well, we might need furher custom ssz types, will likely have to revisit and maybe restructure this at some point